From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: zimzaza4 <3625282098@qq.com>
Date: Fri, 2 Aug 2024 16:36:32 +0800
Subject: [PATCH] NewPathfinderAPI


diff --git a/src/main/java/com/destroystokyo/paper/entity/Pathfinder.java b/src/main/java/com/destroystokyo/paper/entity/Pathfinder.java
index 3c1e2c93d923a683cc0455af77c43784ef12270e..d92089117dde6cad28c82bc2c2c49ac890e41106 100644
--- a/src/main/java/com/destroystokyo/paper/entity/Pathfinder.java
+++ b/src/main/java/com/destroystokyo/paper/entity/Pathfinder.java
@@ -1,10 +1,13 @@
 package com.destroystokyo.paper.entity;
 
 import org.bukkit.Location;
+import org.bukkit.block.Block;
 import org.bukkit.entity.LivingEntity;
 import org.bukkit.entity.Mob;
 
 import java.util.List;
+import java.util.function.Function;
+
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
@@ -179,6 +182,8 @@ public interface Pathfinder {
      */
     void setCanFloat(boolean canFloat);
 
+    void setCanPassHandler(Function<Block, re.imc.marigold.pathfinder.PathType> handler);
+
     /**
      * Represents the result of a pathfinding calculation
      */
diff --git a/src/main/java/re/imc/marigold/pathfinder/PathType.java b/src/main/java/re/imc/marigold/pathfinder/PathType.java
new file mode 100644
index 0000000000000000000000000000000000000000..aae2ab49e4433b33ed4f6f177fe10e5a6cf910e1
--- /dev/null
+++ b/src/main/java/re/imc/marigold/pathfinder/PathType.java
@@ -0,0 +1,7 @@
+package re.imc.marigold.pathfinder;
+
+public enum PathType {
+    OPEN,
+    BLOCKED,
+    DEFAULT
+}
